plugins/LayerView/LayerPass.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) 2016 Ultimaker B.V.
+# Cura is released under the terms of the AGPLv3 or higher.
+
+from UM.Logger import Logger
+from UM.Application import Application
+from UM.Resources import Resources
+from UM.Preferences import Preferences
+from UM.Signal import Signal
+from UM.Job import Job
+
+from UM.Scene.Iterator.DepthFirstIterator import DepthFirstIterator
+from UM.Scene.SceneNode import SceneNode
+from UM.Scene.ToolHandle import ToolHandle
+from UM.Mesh.MeshBuilder import MeshBuilder
+from UM.View.RenderPass import RenderPass
+from UM.View.RenderBatch import RenderBatch
+from UM.View.GL.OpenGL import OpenGL
+
+import numpy
+
+## RenderPass used to display g-code paths.
+class LayerPass(RenderPass):
+    def __init__(self, width, height):
+        super().__init__("layerview", width, height)
+
+        self._controller = Application.getInstance().getController()
+
+        self._shader = None
+        self._tool_handle_shader = None
+        self._gl = OpenGL.getInstance().getBindingsObject()
+        self._scene = self._controller.getScene()
+
+        self._scene.getRoot().childrenChanged.connect(self._onSceneChanged)
+        self._max_layers = 0
+        self._current_layer_num = 0
+        self._current_layer_mesh = None
```

Comment:
I personally think it is nicer to keep the logic for the mesh creation, maximum layer count etc. in LayerView and make the render pass just deal with rendering. Maybe even create a separate class for it, since there is something to be said about it not being the responsibility of LayerView either.